python/pyspark/sql/functions.py

```diff
@@ -2394,6 +2394,23 @@ def array_repeat(col, count):
     return Column(sc._jvm.functions.array_repeat(_to_java_column(col), count))
 
 
+@since(2.4)
+def zip(*cols):
+    """
+    Collection function: Merge two columns into one, such that the M-th element of the N-th
+    argument will be the N-th field of the M-th output element.
+
+    :param cols: columns in input
+
+    >>> from pyspark.sql.functions import zip as spark_zip
```

Comment:
I am not sure about this. What do you think @ueshin ?

Reply:
Hmm, I was thinking about the naming issue actually. Should we use `arrays_zip`, `zip_arrays` or something instead of `zip`? Or any suggestions?

Reply:
I think that we should stick with something related to zip (such as "zip_arrays" or "zip_lists") for "compatibility naming" with other APIs/languages (`Enum.zip` in Elixir and `zip` in Scala, for instance).

Reply:
I think `zip_arrays` should be fine.
Do you have any ideas? @BryanCutler @HyukjinKwon

Reply:
Eh, I think `zip` is fine (like `sum`, `min` or `max`). I agree it's a bad practice to shadow builtin functions but we already started this in this way ...

Reply:
`arrays_zip` is fine to me too, just for clarification if you guys feel hesitant with going ahead as is.

Reply:
Let's use `arrays_zip`.
As for `min`, `max`, we use `array_min` and `array_max` for array type functions apart from aggregate functions. I think `arrays_zip` is consistent with the names.

@DylanGuedes Would you mind if I ask to change to `arrays_zip`? Thanks.

Reply:
Ah! I thought we are talking about shadowing builtin function. Sure, that's better.